src/context/Feed.tsx

```diff
@@ -148,29 +149,31 @@ export function Feed<
     CRT,
     PT
   >();
-  const { feedGroup, userId, children } = props;
+  const { feedGroup, userId, children, options, notify } = props;
   const [, setForceUpdateState] = useState(0);
 
+  const feedId = client?.feed(feedGroup, userId).id;
+
   const manager = useMemo(() => {
-    if (!client) return null;
-    // const clientDifferent = this.props.client !== prevProps.client;
-    // const notifyDifferent = this.props.notify !== prevProps.notify;
-    // const feedDifferent = this.props.userId !== prevProps.userId || this.props.feedGroup !== prevProps.feedGroup;
-    // const optionsDifferent = !_isEqual(this.props.options, prevProps.options);
-    // if (clientDifferent || feedDifferent || optionsDifferent || notifyDifferent)
-    //   TODO: Implement
-    const feedId = client.feed(feedGroup, userId).id;
+    if (!feedId) return null;
+
+    // TODO: check if any of the clients changed
     return (
       sharedFeedManagers[feedId] ||
       new FeedManager<UT, AT, CT, RT, CRT, PT>({ ...props, analyticsClient, client, user, errorHandler })
     );
-  }, [feedGroup, userId]);
+  }, [feedId]);
 
   useEffect(() => {
     const forceUpdate = () => setForceUpdateState((prevState) => prevState + 1);
+    if (manager) manager.props.notify = notify;
     manager?.register(forceUpdate);
     return () => manager?.unregister(forceUpdate);
-  }, [manager]);
+  }, [manager, notify]);
+
+  useDeepCompareEffect(() => {
+    manager?.refresh(options);
```

Comment:
on another thought, this only works the first time. the option that is passed initially are stored in manager and if they change we need to change it within the manager otherwise every single call to refresh need the updated options as param

Reply:
for example you change the options, we correctly update the feed, then you scroll down to load more but we refresh again using the old options

Reply:
True... I'm going to need to add some kind of `updateOptions` to `FeedManager`.